tests/service_tests/job.js

```diff
@@ -615,6 +615,70 @@ exports.setup = function (svc) {
                 );
             });
 
+            it("Callback#Oneshot search with json results", function (done) {
+                var sid = getNextId();
+                var that = this;
+
+                Async.chain([
+                    function (done) {
+                        that.service.jobs().oneshotSearch('search index=_internal | head 1 | stats count', { id: sid, output_mode: 'json' }, done);
+                    },
+                    function (results, done) {
+                        assert.ok(results);

```

Comment:
Can we add some validation of the format coming back? I think ideally it would be clear from the tests what the format coming back is and how to interact with it

Reply:
We are getting response in a **string** format for csv and xml as output_mode.

**XML**: '{"name":"results","value":"","attributes":{"preview":"0"},"children":[{"name":"meta","value":"","attributes":{},"children":[{"name":"fieldOrder","value":"","attributes":{},"children":[{"name":"field","value":"count","attributes":{},"children":[]}]}]},{"name":"result","value":"","attributes":{"offset":"0"},"children":[{"name":"field","value":"","attributes":{"k":"count"},"children":[{"name":"value","value":"","attributes":{},"children":[{"name":"text","value":"1","attributes":{},"children":[]}]}]}]}]}'

**CSV**: "\ncount\n1\n"

Reply:
OK I guess it's a bit unexpected that the XML results are being returned as a JSON string, do you know what's happening there? Is needle or the SDK reformatting the results or are they coming back from Splunk as JSON?

For these tests I think they could be improved by returning multiple events and fields and ensuring that the format matches our expectations. For example:
```javascript
    that.service.jobs().oneshotSearch('makeresults count=3 | streamstats count | eval foo="bar" | fields - _time', { id: sid, output_mode: 'csv' }, done);
                    },
                    function (results, done) {
                        assert.ok(results);
                        assert.strictEqual(results, "count,foo\n1,bar\n2,bar\n3,bar")
```

Reply:
Sure, I'm looking into it and will update you with the findings.

Reply:
As per the suggestion, I have added new assertions to validate the CSV output.